### ISSUE TITLE
Is -pvt=2 like vertical tightness possible without -xlp and/or -nanl?

### ISSUE THREAD

**Issue by rvandam** [perltidy/perltidy#108]
I'm attempting to add perltidy enforcement to a very large codebase with minimal disruption. This has been mostly successful thanks to the wide range of controls. One common pattern I have been unable to account for so far looks like this:

```
$self->call_method(method_name_foo => {
    some_arg1       => $foo,
    some_other_arg3 => $bar->{'baz'},
});
```

where there is a `$method => $args` style "list" of just two items with a break only after the opening brace, not the opening paren.

Default behavior gives me this:
```
$self->call_method(
    method_name_foo => {
        some_arg1       => $foo,
        some_other_arg3 => $bar->{'baz'},
    }
);
```

which honestly doesn't look too bad and we're discussing internally just adopting this pattern instead but it will result in a lot of (a few thousand) changed lines we were hoping to avoid.

I can get a little closer in spirit with -lp -pvt=2:
```
$self->call_method( method_name_foo => {
                                         some_arg1       => $foo,
                                         some_other_arg3 => $bar->{'baz'},
                    }
);
```

but this doesn't really work, both because -lp breaks a lot of other things and because I need -ndnl which conflicts with -lp.

I don't want to use -nanl to avoid the unwanted newline because that breaks -wn which is immensely valuable elsewhere (and because generally we don't object to _more_ newlines, just in this one use case).

Any ideas on how this might be possible? I had hoped that -wn would rescue me but it seems to have no effect here, presumably because the intervening list caused this not to be treated as "nested containers". I've also tried all values of -cab to no avail.

I'm hoping there's an existing combination I'm missing but if not, perhaps a new controller/inclusion list to allow short lists to nest with -wn could work? or alternatively perhaps some variation on -cab like --brace-arrow-breakpoint=2?

Current options dump:
```
--add-newlines
--noadd-semicolons
--add-terminal-newline
--add-whitespace
--backup-file-extension="bak"
--blank-lines-before-packages=1
--blank-lines-before-subs=1
--blanks-before-blocks
--blanks-before-comments
--block-brace-tightness=0
--block-brace-vertical-tightness=2
--block-brace-vertical-tightness-list="*"
--nobrace-left-and-indent
--brace-tightness=1
--brace-vertical-tightness=0
--brace-vertical-tightness-closing=0
--break-after-labels=0
--break-at-old-attribute-breakpoints
--break-at-old-comma-breakpoints
--break-at-old-keyword-breakpoints
--break-at-old-logical-breakpoints
--break-at-old-method-breakpoints
--break-at-old-ternary-breakpoints
--break-before-all-operators
--break-before-hash-brace=0
--break-before-hash-brace-and-indent=0
--break-before-paren=0
--break-before-paren-and-indent=0
--break-before-square-bracket=0
--break-before-square-bracket-and-indent=0
--character-encoding="guess"
--nocheck-syntax
--closing-brace-indentation=0
--closing-paren-indentation=0
--closing-side-comment-else-flag=0
--closing-side-comment-interval=6
--closing-side-comment-maximum-text=20
--closing-side-comments-balanced
--closing-square-bracket-indentation=0
--code-skipping
--comma-arrow-breakpoints=0
--continuation-indentation=4
--cuddled-break-option=1
--cuddled-else
--default-tabsize=8
--nodelete-old-newlines
--nodelete-old-whitespace
--delete-semicolons
--dump-options
--encode-output-strings
--extended-continuation-indentation
--extended-syntax
--format="tidy"
--format-skipping
--function-paren-vertical-alignment
--fuzzy-line-length
--nohanging-side-comments
--nohtml
--html-entities
--html-table-of-contents
--ignore-side-comment-lengths
--indent-block-comments
--indent-columns=4
--iterations=1
--keep-old-blank-lines=1
--keyword-group-blanks-after=1
--keyword-group-blanks-before=1
--nokeyword-group-blanks-delete
--nokeyword-group-blanks-inside
--keyword-group-blanks-repeat-count=0
--keyword-group-blanks-size=5
--keyword-paren-inner-tightness=1
--nologfile
--logical-padding
--long-block-line-count=8
--look-for-autoloader
--look-for-selfloader
--maximum-consecutive-blank-lines=1
--maximum-fields-per-table=4
--maximum-file-size-mb=10
--maximum-level-errors=1
--maximum-line-length=150
--maximum-unexpected-errors=0
--memoize
--minimum-space-to-comment=1
--non-indenting-braces
--nowant-right-space="A"
--one-line-block-nesting=0
--one-line-block-semicolons=1
--opening-brace-always-on-right
--opening-hash-brace-right
--opening-paren-right
--opening-square-bracket-right
--nooutdent-labels
--nooutdent-long-comments
--nooutdent-long-quotes
--output-line-ending="unix"
--paren-tightness=2
--paren-vertical-tightness=0
--paren-vertical-tightness-closing=0
--pass-version-line
--perl-syntax-check-flags="-c -T"
--pod2html
--noquiet
--recombine
--short-concatenation-item-length=8
--noshow-options
--space-backslash-quote=1
--nospace-for-semicolon
--space-prototype-paren=1
--nospace-terminal-semicolon
--square-bracket-tightness=2
--square-bracket-vertical-tightness=0
--square-bracket-vertical-tightness-closing=0
--stack-opening-hash-brace
--stack-opening-paren
--stack-opening-square-bracket
--standard-error-output
--standard-output
--static-block-comment-prefix="#"
--static-block-comments
--static-side-comment-prefix="#"
--static-side-comments
--notabs
--tight-secret-operators
--timestamp
--notrim-qw
--nouse-unicode-gcstring
--valign-block-comments
--valign-code
--valign-side-comments
--want-left-space="A"
--want-right-space="!"
--nowarning-output
--weld-nested-containers
```

**Comment by shancock9** [comment]
I'll take a look. I think I can tweak the weld logic to handle this.

**Comment by shancock9** [comment]
A new option -wfc (--weld-fat-comma) has been tentatively added which allows welding to occur when there is a single hash key and '=>' between an opening paren and any other opening type. So for example:

```
        # perltidy -wn -wfc
        elf->call_method( method_name_foo => {
            some_arg1       => $foo,
            some_other_arg3 => $bar->{'baz'},
        } );
```

This update worked well when tested on a variety of code. It is available in tagged version 20220613.05 :

https://github.com/perltidy/perltidy/releases/tag/20220613.05

The next release to CPAN should be in about a month.

**Comment by shancock9** [comment]
Version 20221112 has this update.
